platform/rct/hal/ADC_stm32/ADC_device_stm32.c

```diff
@@ -76,54 +79,44 @@ int ADC_device_init(void)
         GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_NOPULL;
         GPIO_Init(GPIOB, &GPIO_InitStructure);
 
-        /* Calibration procedure
-         *  TODO see if this is necessary
-         *
-         ADC_VoltageRegulatorCmd(ADC1, ENABLE);
-
-         ADC_SelectCalibrationMode(ADC1, ADC_CalibrationMode_Single);
-         ADC_StartCalibration(ADC1);
-
-         while(ADC_GetCalibrationStatus(ADC1) != RESET );
-         calibration_value = ADC_GetCalibrationValue(ADC1);
-        */
-
-        ADC_CommonInitTypeDef   ADC_CommonInitStructure;
-
-        ADC_CommonInitStructure.ADC_Mode = ADC_Mode_Independent;
-        ADC_CommonInitStructure.ADC_Clock = ADC_Clock_AsynClkMode;
-        ADC_CommonInitStructure.ADC_DMAAccessMode = ADC_DMAAccessMode_Disabled;
-        ADC_CommonInitStructure.ADC_DMAMode = ADC_DMAMode_Circular;
-        ADC_CommonInitStructure.ADC_TwoSamplingDelay = 0;
-        ADC_CommonInit(ADC3, &ADC_CommonInitStructure);
-
-        ADC_InitTypeDef         ADC_InitStructure;
+        /* Calibration procedure */
+        ADC_VoltageRegulatorCmd(ADC3, ENABLE);
+        /*
+         * Wait at least 10us before starting calibration or enabling.
+         * This is a total hack since I don't want to put a timer here
+         * and there is no interrupt to tell me this is ready :(.
+         */
+        for (size_t i = 0; i < 10000; ++i);
```

Comment:
Good call.  I was still running under the assumption that it would be possible to this to be invoked before the scheduler was called.  If that is still possible then no, else yes.